tests/lib/rules/valid-slot-scope.js

```diff
@@ -0,0 +1,393 @@
+              {{a}}
+            </div>
+          </MyComponent>
+        </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope="[a, ...b]">
+              {{a}}
+            </div>
+          </MyComponent>
+        </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope="{a, ...b}">
+              {{a}}
+            </div>
+          </MyComponent>
+        </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope="     slotProps     ">
+              {{slotProps.name}}
+            </div>
+          </MyComponent>
+        </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope=" /*comment*/ slotProps /*comment*/ ">
+              {{slotProps.name}}
+            </div>
+          </MyComponent>
+        </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <template scope="slotProps">
+              <div>
+                {{slotProps.name}}
+              </div>
+            </template>
+          </MyComponent>
+        </template>
+      `
+    }
+  ],
+  invalid: [
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope />
+          </MyComponent>
+        </template>
+      `,
+      errors: [{
+        message: "'slot-scope' attributes require a value.",
+        line: 4,
+        column: 18,
+        endColumn: 28
+      }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope="" />
+          </MyComponent>
+        </template>
+      `,
+      errors: [{
+        message: "'slot-scope' attributes require a value.",
+        line: 4
+      }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <template scope="">
+              <div />
+            </template>
+          </MyComponent>
+        </template>
+      `,
+      errors: [{
+        message: "'scope' attributes require a value.",
+        line: 4
+      }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope="...args">
+              {{args}}
+            </div>
+          </MyComponent>
+        </template>
+      `,
+      errors: [{
+        message: 'The top level rest parameter is useless.',
+        line: 4,
+        column: 30,
+        endColumn: 37
+      }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <MyComponent>
+            <div slot-scope="a, b  ">
```

Comment:
I'm surprised in this is not a syntax error.
Does not `no-parsing-error` catch this? If so, it looks like a bug on `vue-eslint-parser`.

Reply:
It was that. Now I remember that I made the parser's behavior same as the original compiler.

But it's odd if orphan tokens exist without syntax errors, so I will add the syntax error into `vue-eslint-parser`.

Reply:
After updating `vue-eslint-parser`, I would like to update this rule.

Reply:
I have released `vue-eslint-parser` [v3.3.0](https://github.com/mysticatea/vue-eslint-parser/releases/tag/v3.3.0).